pycrunch/shoji.py

```diff
@@ -206,6 +205,27 @@ def replace(self, entity=None):
             entity = self
         return super(Entity, self).put(data=entity.json).payload
 
+    def wait_progress(self, r, timeout=None):
+        self.self = URL(r.headers['Location'], '')
+        progress_url = r.payload['value']
+
+        begin = time.time()
+        while timeout is None or time.time() - begin < timeout:
+            prog_r = self.session.get(progress_url)
+            progress = prog_r.payload['value']
+            if progress['progress'] == -1:
+                # Completed due to error
+                raise TaskError(progress['message'])
+            elif progress['progress'] == 100:
+                # Completed with success
+                break
+            time.sleep(0.5)
```

Comment:
`wait_for_batch_status` had `backoff_rate` and `backoff_max` args: should those be built in here?